fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/api/JDBCIdentityCache.java

```diff
@@ -69,4 +79,10 @@
      * @return
      */
     List<Long> getCommonTokenValueIdList(String tokenValue);
+
+    /**
+     * Get the set of all resource type names.
+     * @return
+     */
+    Set<String> getResourceTypeNames() throws FHIRPersistenceException;
```

Comment:
In ModelSupport we have a similar method which takes an argument for whether or not you want it to include "abstract" types.
At the very least I think we should clearly document whether this list will include those types or not (seems like yes at the moment, but should it?)

Reply:
We should fix this when we properly implement support for a subset of resources (so that the whole-system search union only selects from the tables which are actually provisioned in the schema).

Reply:
Added as a comment to https://github.com/IBM/FHIR/issues/819#issuecomment-868725173 but I still think we should update the javadoc to indicate that it current returns abstract types in the meantime.